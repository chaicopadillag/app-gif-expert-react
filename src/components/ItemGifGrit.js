import React from 'react';
import PropTypes from 'prop-types';

const ItemGifGrit = ({ title, url }) => {
	return (
		<div className="card animate__animated animate__fadeIn">
			<img src={url} alt={title} />
			<p>{title}</p>
		</div>
	);
};

ItemGifGrit.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};
export default ItemGifGrit;
