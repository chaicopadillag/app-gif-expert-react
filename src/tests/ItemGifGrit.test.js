import React from 'react';
import { shallow } from 'enzyme';
import ItemGifGrit from '../components/ItemGifGrit';

describe('Prueba de <ItemGifGrit/>', () => {
	const props = {
		title: 'Mi Titulo',
		url: 'https://localhost/img.jpg',
	};
	const wrapper = shallow(<ItemGifGrit {...props} />);

	test('Debe mostrar el componente <ItemGifGrit/>', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('evaluar titulo enviado', () => {
		const titulo = wrapper.find('p').text().trim();
		expect(titulo).toBe(props.title);

		const img = wrapper.find('img');
		expect(img.props().src).toBe(props.url);
	});

	test('debe de tener la clase de animate_fadeIn', () => {
		const div = wrapper.find('div');
		expect(div.props().className).toContain('animate__fadeIn');
	});
});
