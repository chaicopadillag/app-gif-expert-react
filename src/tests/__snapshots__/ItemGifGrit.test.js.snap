// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <ItemGifGrit/> Debe mostrar el componente <ItemGifGrit/> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Mi Titulo"
    src="https://localhost/img.jpg"
  />
  <p>
    Mi Titulo
  </p>
</div>
`;
